sdk/src/main/java/io/radar/sdk/util/RadarSimpleLogBuffer.kt

```diff
@@ -3,59 +3,189 @@ package io.radar.sdk.util
 import io.radar.sdk.Radar
 import io.radar.sdk.model.RadarLog
 import java.util.concurrent.LinkedBlockingDeque
+import android.content.Context
+import io.radar.sdk.RadarSettings
+import org.json.JSONObject
+import java.io.File
+import java.lang.Integer.min
+import java.util.Date
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
 
 /**
- * Basic Log Buffer implementation that is backed by a field.
+ * Log Buffer implementation that persist.
  */
-internal class RadarSimpleLogBuffer : RadarLogBuffer {
+internal class
+RadarSimpleLogBuffer(override val context: Context): RadarLogBuffer{
 
     private companion object {
-        const val MAXIMUM_CAPACITY = 1000
+
+        const val MAXIMUM_MEMORY_CAPACITY = 200
+
+        const val MAXIMUM_CAPACITY = 500
         const val HALF_CAPACITY = MAXIMUM_CAPACITY / 2
+        const val logFileDir = "radar_logs"
+        var counter = 0
+
     }
+    private var featureFlag = false
 
-    /**
-     * Concurrency-safe list of logs with a maximum capacity of 1000
-     */
-    private val list = LinkedBlockingDeque<RadarLog>(MAXIMUM_CAPACITY)
 
-    override fun write(level: Radar.RadarLogLevel, message: String, type: Radar.RadarLogType?) {
-        if (!list.offer(RadarLog(level, message, type))) {
-            purgeOldestLogs()
-            list.put(RadarLog(level, message, type))
+    private val scheduler = Executors.newScheduledThreadPool(1)
+
+    init {
+        scheduler.scheduleAtFixedRate({ persistLogs() }, 30, 30, TimeUnit.SECONDS)
+        val file = File(context.filesDir, logFileDir)
+        if (!file.exists()) {
+            file.mkdir()
         }
+        featureFlag = RadarSettings.getFeatureSettings(context).useLogPersistence
+    }
+
+    private val list = LinkedBlockingDeque<RadarLog>(MAXIMUM_MEMORY_CAPACITY)
+    private val oldList = LinkedBlockingDeque<RadarLog>(MAXIMUM_CAPACITY)
```

Comment:
why do we need two lists?

Reply:
I'm currently maintaining the old implementation alongside the new one as we roll it out slowly. Not sharing a list ensures makes it easier to reason about each implementation of logBuffer separately.

Reply:
It's kindof hard to follow the logic with both lists - I have a feeling it will be easier to get right with one list but maybe I'm missing something

Reply:
@lmeier @KennyHuRadar I think it's worth doing a spike on this that only uses one list - even if there are a few compromises. At least to see what it looks like.